File: LeetCode/removeElement.js
const removeElement = (nums, val) => {
  let ct = 0;
  const asObj = {};
  const len = nums.length;
  for (let i = 0; i < len; i++) {
    const num = nums[i];
    if (num === val || asObj[i]) {
      ct++;
      let notFound = true;
      for (let j = i + 1; j < len; j++) {
        const numNew = nums[j];
        if (numNew !== val && !asObj[j]) {
          nums[i] = numNew;
          asObj[j] = true;
          notFound = false;
          break;
        }
      }
      if (notFound) {
        return i;
      }
    }
  }
};

File: LeetCode/DailyDoseCSharp/RemoveElement.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        var i = 0;
        var j = nums.Length - 1;
        if (j <= 0)
        {
            return  j < 0 || nums[0] == val ? 0 : 1;
        }
        while (i < j)
        {
            if (nums[i] == val)
            {
                while (nums[j] == val)
                {
                    if (i == j)
                    {
                        return i;
                    }
                    j--;
                }
                nums[i] = nums[j];
                nums[j] = val;
            }
            i++;
        }
        return nums[i] == val ? i : i + 1;
    }
}
